Pinta/Dialogs/CanvasGridSettingsDialog.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using Gtk;
+using Pinta.Core;
+using Pinta.Core.Managers;
+
+namespace Pinta;
+
+public sealed class CanvasGridSettingsDialog : Dialog
+{
+	private readonly CanvasGridManager canvas_grid;
+
+	private readonly bool initial_show_grid;
+	private readonly int initial_grid_width;
+	private readonly int initial_grid_height;
+
+	private readonly CheckButton show_grid_checkbox;
+	private readonly SpinButton grid_width_spinner;
+	private readonly SpinButton grid_height_spinner;
+
+	private const int Spacing = 6;
+
+	internal CanvasGridSettingsDialog (
+		ChromeManager chrome,
+		CanvasGridManager canvasGrid)
+	{
+		canvas_grid = canvasGrid;
+
+		initial_show_grid = canvas_grid.ShowGrid;
+		initial_grid_width = canvas_grid.CellWidth;
+		initial_grid_height = canvas_grid.CellHeight;
+
+		show_grid_checkbox = CreateShowGridCheckbox (initial_show_grid, SettingsChanged);
+		grid_width_spinner = CreateSpinner (initial_grid_width, SettingsChanged);
+		grid_height_spinner = CreateSpinner (initial_grid_height, SettingsChanged);
+
+		Grid grid = new () {
+			RowSpacing = Spacing,
+			ColumnSpacing = Spacing,
+			ColumnHomogeneous = false,
+		};
+
+		grid.Attach (show_grid_checkbox, 0, 0, 2, 1);
+
+		grid.Attach (CreateLabel ("Width:", Align.End), 0, 1, 1, 1);
```

Comment:
The call to `Translations.GetString()` actually needs to happen here around the string literal, e.g. `Translations.GetString("Width:")`, rather than inside the `CreateLabel()` method - otherwise these strings won't be picked up with the automated tool that finds all the translatable strings (https://github.com/PintaProject/Pinta/wiki/Translations)

(Same for the `Height:` label below)